src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/PreferHashDataOverComputeHash.cs

```diff
@@ -0,0 +1,339 @@
+                {
+                    switch (computeHashInvocation.Instance)
+                    {
+                        case ILocalReferenceOperation localReferenceOperation:
+                            computeHashVariableMap.TryAdd(localReferenceOperation, computeHashInvocation);
+                            break;
+                        case IInvocationOperation chainedInvocationOperation when IsHashCreateMethod(chainedInvocationOperation.TargetMethod):
+                            ReportChainedComputeHashInvocationOperation(context, computeHashInvocation, chainedInvocationOperation.TargetMethod.ContainingType, byteArrayParameter);
+                            break;
+                        case IObjectCreationOperation chainObjectCreationOperation when chainObjectCreationOperation.Type.Inherits(hashAlgoBaseType):
+                            ReportChainedComputeHashInvocationOperation(context, computeHashInvocation, chainObjectCreationOperation.Type, byteArrayParameter);
+                            break;
+                    }
+                }
+
+                void OnOperationBlockEnd(OperationBlockAnalysisContext context)
+                {
+                    var singleLocalRefElements = localReferenceMap
+                        .GroupBy(local => local.Value)
+                        .Where(local => local.HasExactly(1))
+                        .Select(local => local.First().Key);
+
+                    foreach (var localRef in singleLocalRefElements)
+                    {
+                        if (!createdSymbolMap.TryGetValue(localRef.Local, out var declarationTuple) ||
+                            !computeHashVariableMap.TryGetValue(localRef, out var computeHashMethod) ||
+                            !TryGetHashDataMethod(declarationTuple.OriginalType, byteArrayParameter, out var staticHashMethod))
+                        {
+                            continue;
+                        }
+
+                        var codefixerLocations = GetFixerLocations(declarationTuple.DeclaratorOperation, computeHashMethod);
+                        var diagnostics = CreateDiagnostics(computeHashMethod, staticHashMethod.ContainingType, codefixerLocations);
+                        context.ReportDiagnostic(diagnostics);
+                    }
+
+                    computeHashVariableMap.Free(context.CancellationToken);
+                    createdSymbolMap.Free(context.CancellationToken);
+                    localReferenceMap.Free(context.CancellationToken);
+                }
+
+                bool IsHashCreateMethod(IMethodSymbol methodSymbol)
+                {
+                    return methodSymbol.ContainingType.Inherits(hashAlgoBaseType) &&
+                        methodSymbol.ReturnType.Inherits(hashAlgoBaseType) &&
+                        methodSymbol.Name.Equals(CreateMethodName, StringComparison.Ordinal);
```

Comment:
The value of `CreateMethodName` is `System.Security.Cryptography.SHA256.Create` isn't it mean it will not work for the other hash algorithms?

Reply:
The value is `Create`

Reply:
Right forgot that 😆